### DISCUSSION TITLE
Make security rules part of the framework

### DISCUSSION THREAD

**Discussion by dkbast** [serverpod/serverpod#2065] [Ideas]
Many Flutter developers will be coming from Firebase with security rules. 

I would like to suggest something that makes working with "per user" data easier (tables with a "userId" column).

It would be great if we could mark the generated models as "user bound" which creates a field for the user id - the generated database connection could then take the userId from the session into account when querying data, reducing potential errors. 

For queries over all data from multiple user ids there should be a way which makes it explicit that you need to know what you are doing :)

**Comment by SandPod** [comment]
Interesting thought!
I agree that simplifying and automating row-based rules would be really helpful for developers.

Feel free to flesh this idea out a bit and turn it into a feature request.
It would be great if you could include your thoughts on the syntax in the model files and the query interface to the database.
